GithubIssueTagger/Reports/IceBox/IceBoxReport.cs

```diff
@@ -0,0 +1,351 @@
+                            _addLabelId = addLabel.Id;
+                        }
+
+                        // action label already applied, skip
+                        continue;
+                    }
+                    else if (issue.Labels.PageInfo.HasNextPage)
+                    {
+                        // TODO: Handle when issue has more than 100 labels
+                        Console.WriteLine("##vso[task.logissue type=warning]Unsupported scenario: issue " + issue.Number + " has more than 100 labels");
+                    }
+                }
+
+                if (!TryGetLastLabelTime(issue.TimelineItems.Nodes, label, out DateTime? labelAdded))
+                {
+                    labelAdded = await GetLastLabelTimeAsync(issue.Id, label);
+                    if (labelAdded == null)
+                    {
+                        // TODO: If we reach here, we need to do a different GraphQL query to get more events for this issue to find the last time the label was added.
+                        Console.WriteLine("##vso[task.logissue type=warning]Unsupported scenario: issue " + issue.Number + " did not find label in latest events");
+                        continue;
+                    }
+                }
+
+                if (HasEnoughPositiveReactions(issue, labelAdded.Value, upvoteCount))
+                {
+                    Console.WriteLine($"Issue {issue.Number} has enough upvotes");
+                    if (add != null)
+                    {
+                        if (_addLabelId == null)
+                        {
+                            _addLabelId = await GetLabelIdAsync(owner, repo, add);
+                        }
+
+                        await AddLabelToIssueAsync(issue.Id, _addLabelId);
+                    }
+                }
+            }
+        }
+
+        private async IAsyncEnumerable<GetIssuesResult.IssuesModel> GetIssuesAsync(string owner, string repo, string label, int upvotes)
+        {
+            // See GitHub docs on resource/query limits. Increasing the counts has a multiplactive effect towards the hourly query limit.
+            Dictionary<string, object?>? variables = new Dictionary<string, object?>()
+            {
+                ["owner"] = owner,
+                ["repo"] = repo,
+                ["after"] = null,
+                ["label"] = label,
+                ["timelineCount"] = 5, // adding icebox label is usually one of the most recent actions
+                ["reactionCount"] = upvotes * 2
+            };
+
+            var request = new GraphQLRequest(IceBoxResource.GetIssues)
+            {
+                Variables = variables
+            };
+
+            while (variables != null)
+            {
+                GraphQLResponse<GetIssuesResult>? response = await _client.SendAsync<GetIssuesResult>(request);
+                if (response?.Data?.Repository.Issues.Nodes != null)
```

Comment:
Ideally, yes. I was focusing on the happy path and error handling happened when I remembered, but is basically untested because I validated the graphql queries externally before copying the text here 😳.  That is to say, the risk of error is very low, so I'm wondering how much effort is worthwhile given this is a non-shipping, internal tool.

Reply:
Okay sounds good